pytest_motor/plugin.py

```diff
@@ -51,45 +52,36 @@ async def root_directory(pytestconfig: PytestConfig) -> Path:
 async def mongod_binary(root_directory: Path) -> Path:  # pylint: disable=redefined-outer-name
     """Return a path to a mongod binary."""
     destination: Path = root_directory.joinpath('.mongod')
-    mongod_binary_filename = destination.joinpath(
-        'mongodb-linux-x86_64-ubuntu1804-4.4.6/bin/mongod')
+    mongod_binary_relative_path = 'mongodb-linux-x86_64-ubuntu1804-4.4.6/bin/mongod'
+    mongod_binary_filename = destination.joinpath(mongod_binary_relative_path)
 
     if not mongod_binary_filename.exists():
         download_url = 'https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1804-4.4.6.tgz'
-        (tar_filename, _) = urllib.request.urlretrieve(download_url)
-
-        with tarfile.open(tar_filename) as tar:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(download_url) as resp:
+                with open('binaries.tar', 'wb') as binary_file:
+                    # Read by chunks to avoid big RAM consumption
+                    while True:
+                        # read by 100 bytes
+                        chunk = await resp.content.read(100)
+                        if not chunk:
+                            break
+                        binary_file.write(chunk)
+
+        with tarfile.open("binaries.tar") as tar:
             tar.extract(member='mongodb-linux-x86_64-ubuntu1804-4.4.6/bin/mongod', path=destination)
 
-        mongod_binary_filename = destination.joinpath(
-            'mongodb-linux-x86_64-ubuntu1804-4.4.6/bin/mongod')
-
     return mongod_binary_filename
 
 
 @pytest.fixture(scope='function')
-# pylint: disable=redefined-outer-name
-async def sockets_directory(root_directory: Path) -> AsyncIterator[Path]:
-    """Yield a directory for MongodDB unix sockets."""
-    sockets_directory = root_directory.joinpath('.mongod_sockets')
-    sockets_directory.mkdir(exist_ok=True)
-
-    yield sockets_directory
-
-
-@pytest.fixture(scope='function')
-# pylint: disable=redefined-outer-name
-async def unix_socket(sockets_directory: Path) -> AsyncIterator[Path]:
-    """Yield a random unix socket in the given sockets directory."""
-    name: str = secrets.token_hex(12)
-    unix_socket = sockets_directory.joinpath(f'{name}.sock')
-
-    yield unix_socket
-
-    try:
-        unix_socket.unlink()
-    except FileNotFoundError:  # pragma: no cover
-        pass
+def new_port() -> int:
+    """Return an unused port for mongod to run on."""
+    port: int = 27017
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as opened_socket:
+        opened_socket.bind(("127.0.0.1", 0))  # system will automaticly assign port
+        port = opened_socket.getsockname()[1]
+    return port
```

Comment:
I think we probably want to yield the port inside of the context manager. According to the [socket object docs](https://docs.python.org/3/library/socket.html#socket-objects), "Exiting the context manager is equivalent to calling `close()`".

Reply:
I don't think that's a good idea. If I understand correctly, this means that python will reserve this socket, and mongod won't bind it, because it's already allocated.

Reply:
Hmm okay. So we probably don't want to do that then, but but should we still perform some cleanup afterwards?

Reply:
What do you mean by cleanup? Mongod will free the port as it closes, won't it?

Reply:
I wasn't sure if it does or not?

Reply:
I think it does. Will write a test for it.

Reply:
Added this test and it passes! 🎉 

```python
@pytest.mark.asyncio
async def test_port_not_allocated(mongod_binary: Path, new_port: int) -> None:
    """Tests that mongod leaves port free"""
    mongod = await asyncio.create_subprocess_exec(*[str(mongod_binary), '--port', str(new_port)])
    mongod.terminate()
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as opened_socket:
        assert opened_socket.connect_ex(('127.0.0.1', new_port)) != 0
```